opnode/node/node.go

```diff
@@ -138,27 +126,28 @@ func (c *OpNode) Start() error {
 	c.log.Info("Attaching execution engine(s)")
 	for _, eng := range c.l2Engines {
 		// Request initial head update, default to genesis otherwise
-		reqCtx, reqCancel := context.WithTimeout(c.ctx, time.Second*10)
-		if !eng.RequestUpdate(reqCtx, eng.Log, eng) {
-			eng.Log.Error("failed to fetch engine head, defaulting to genesis")
-			eng.UpdateHead(eng.Genesis.L1, eng.Genesis.L2)
-		}
-		reqCancel()
+		reqCtx, reqCancel := context.WithTimeout(ctx, time.Second*10)
 
 		// driver subscribes to L1 head changes
 		l1SubCh := make(chan eth.HeadSignal, 10)
 		l1HeadsFeed.Subscribe(l1SubCh)
 		// start driving engine: sync blocks by deriving them from L1 and driving them into the engine
-		engDriveSub := eng.Drive(c.ctx, l1SubCh)
-		handleUnsubscribe(engDriveSub, "engine driver unexpectedly failed")
+		err := eng.Start(reqCtx, l1SubCh)
+		// engDriveSub := eng.Drive(c.ctx, l1SubCh)
+		// handleUnsubscribe(engDriveSub, "engine driver unexpectedly failed")
+		reqCancel()
+		if err != nil {
+			c.log.Error("Could not start a rollup node", "err", err)
+			return err
+		}
 	}
 
 	// Keep subscribed to the L1 heads, which keeps the L1 maintainer pointing to the best headers to sync
 	l1HeadsSub := event.ResubscribeErr(time.Second*10, func(ctx context.Context, err error) (event.Subscription, error) {
 		if err != nil {
 			c.log.Warn("resubscribing after failed L1 subscription", "err", err)
```

Comment:
```suggestion
		if err != nil {
			if err == ctx.Err() {
				return nil, err
			}
			c.log.Warn("resubscribing after failed L1 subscription", "err", err)
```
If the outer ctx fails, we shouldn't create the new subscription